test/uk/gov/hmrc/plasticpackagingtax/registration/controllers/group/GroupMemberGrsControllerSpec.scala

```diff
@@ -0,0 +1,130 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.plasticpackagingtax.registration.controllers.group
+
+import base.unit.ControllerSpec
+import org.mockito.ArgumentMatchers.any
+import org.mockito.Mockito.verify
+import org.mockito.{ArgumentCaptor, Mockito}
+import org.scalatest.matchers.must.Matchers.convertToAnyMustWrapper
+import play.api.http.Status.SEE_OTHER
+import play.api.test.Helpers.{await, redirectLocation, status}
+import uk.gov.hmrc.http.InternalServerException
+import uk.gov.hmrc.plasticpackagingtax.registration.connectors.DownstreamServiceError
+import uk.gov.hmrc.plasticpackagingtax.registration.forms.OrgType
+import uk.gov.hmrc.plasticpackagingtax.registration.models.registration.{
+  GroupDetail,
+  OrganisationDetails,
+  Registration
+}
+import uk.gov.hmrc.play.bootstrap.tools.Stubs.stubMessagesControllerComponents
+
+class GroupMemberGrsControllerSpec extends ControllerSpec {
+
+  private val mcc          = stubMessagesControllerComponents()
+  private val registration = aRegistration()
+
+  private val controller =
+    new GroupMemberGrsController(authenticate = mockAuthAction,
+                                 mockJourneyAction,
+                                 mockRegistrationConnector,
+                                 mockUkCompanyGrsConnector,
+                                 mcc
+    )(ec)
+
+  private val registeredLimitedCompany = aRegistration(
+    withGroupDetail(
+      Some(
+        GroupDetail(membersUnderGroupControl = Some(true),
+                    members = Seq.empty,
+                    currentMemberOrganisationType = Some(OrgType.UK_COMPANY)
+        )
+      )
+    )
+  )
+
+  "group member incorpIdCallback " should {
+    "redirect to the group member organisation details type controller page" when {
+      "registering a UK Limited Company" in {
+        val result = simulateLimitedCompanyCallback()
+
+        status(result) mustBe SEE_OTHER
+        redirectLocation(result) mustBe Some(routes.OrganisationListController.displayPage().url)
+      }
+
+    }
+
+    "store the returned business entity information" when {
+      "registering UK Limited Company group member" in {
+        await(simulateLimitedCompanyCallback())
+
+        val memberDetails = getLastSavedRegistration.groupDetail.get.members.last
+
+        memberDetails.organisationDetails mustBe groupMemberDetailsUKCompany.organisationDetails
+        memberDetails.addressDetails mustBe groupMemberDetailsUKCompany.addressDetails
+      }
```

Comment:
This test is pretty difficult to reason about. It relies on the fact that test data defined elsewhere is consistent; essentially that `PptTestData.groupMemberDetailsUKCompany` and `PptTestData.groupMemberDetailsUKCompany` are built with the same pieces of test data. 

I think we could improve tests of this nature by making it clearer to the reader that what's being checked is the same thing. In this case, perhaps, by constructing the group member details in the test from constituent parts of the `incorporationDetails` which is used by the involved mocks.

@kiranprakashdigital please don't see this as a criticism. This is a pet hate of mine and I know there are other examples elsewhere and, very probably, bad examples written by me.

Reply:
I have changed the test to compare to `incorporaationDetails` companyName and addressDetails.